antsibull/write_docs.py

```diff
@@ -135,12 +142,15 @@ async def output_all_plugin_rst(collection_info: CollectionInfoT,
     writers = []
     lib_ctx = app_context.lib_ctx.get()
     async with asyncio_pool.AioPool(size=lib_ctx.thread_max) as pool:
-        for collection_name, plugins_by_type in collection_info.items():
+        for collection_name, plugins_by_type in collection_to_plugin_info.items():
             for plugin_type, plugins in plugins_by_type.items():
                 for plugin_short_name, dummy_ in plugins.items():
                     plugin_name = '.'.join((collection_name, plugin_short_name))
                     writers.append(await pool.spawn(
-                        write_rst(collection_name, plugin_short_name, plugin_type,
+                        write_rst(collection_name,
+                                  collection_metadata.get(collection_name) or
+                                  AnsibleCollectionMetadata.empty(),
```

Comment:
Is there ever a time when `collection_metadata[collection_name]` does not exist?  It feels like `AnsibleCollectionMetadata.empty()` should be filled in earlier (maybe in `augment_docs()` except that's keyed by plugin so it doesn't make a ton of sense there.  Or maybe in `get_collection_contents()` but then the plugin-view and collection-view don't match.)  If `collection_metadata[collection_name]` always exists then we can sidestep that decision by just removing the `or AnsibleCollectionMetadata.empty()` from here.

Reply:
Hopefully not, except if there's a bug somewhere. It's annoying to have to go looking for a bug if you want to release a new version when you're usually busy with other things :) If you prefer a crash here if something went wrong, I can change it.

Reply:
Changed in 9f0a1caba53382cc080f9f33afd0d5c659274ec3.